protocols/identify/src/behaviour.rs

```diff
@@ -445,42 +458,31 @@ fn multiaddr_matches_peer_id(addr: &Multiaddr, peer_id: &PeerId) -> bool {
     true
 }
 
-struct PeerCache(Option<LruCache<PeerId, HashSet<Multiaddr>>>);
+struct PeerCache(Option<PeerAddresses>);
 
 impl PeerCache {
     fn disabled() -> Self {
         Self(None)
     }
 
     fn enabled(size: NonZeroUsize) -> Self {
-        Self(Some(LruCache::new(size)))
-    }
-
-    fn get_mut(&mut self, peer: &PeerId) -> Option<&mut HashSet<Multiaddr>> {
-        self.0.as_mut()?.get_mut(peer)
+        Self(Some(PeerAddresses::new(size)))
     }
 
     fn put(&mut self, peer: PeerId, addresses: impl Iterator<Item = Multiaddr>) {
-        let cache = match self.0.as_mut() {
-            None => return,
-            Some(cache) => cache,
-        };
```

Comment:
I'd prefer to stick to the original early-exit design.

1. It reduces the diff here.
2. It special-cases the non-happy path via early exit and keeps the happy path on the first level of indentation of the function.

Reply:
:heavy_check_mark: